pkg/provisioner/ironic/ironic.go

```diff
@@ -1115,11 +1134,12 @@ func (p *ironicProvisioner) Deprovision() (result provisioner.Result, err error)
 		"current", ironicNode.ProvisionState,
 		"target", ironicNode.TargetProvisionState,
 		"deploy step", ironicNode.DeployStep,
+		"instance_info", ironicNode.InstanceInfo,
 	)
 
 	switch nodes.ProvisionState(ironicNode.ProvisionState) {
 
-	case nodes.Error, nodes.CleanFail:
+	case nodes.Error, nodes.CleanFail, nodes.AdoptFail:
 		if !ironicNode.Maintenance {
```

Comment:
This whole part is completely wrong, let's not add to it. You cannot use `deleted` in `adopt failed`, and you don't need maintenance except when removing an active node.

Reply:
How do we get a node out of "Adopt Failed"?

Reply:
`manage` to go to `manageable` (unless you want to retry adoption).

Reply:
Related: #697

Reply:
@zaneb do we have to take into account recovery from "Adopt Failed "  while deprovisioning as well? What I had in mind was we just want to recover from "Adopt Failed "  while we are trying to adopt..

Reply:
We also need to adopt if we find the node missing during deprovisioning, which is what this PR is about.

Reply:
Thanks for confirmation @zaneb